### PR TITLE
Sticky the Misc scenery tab next to the All scenery tab

### PR DESCRIPTION
**Pull request by fidwell**
During some discussion with folks on the NE Discord server, it was suggested  to anchor the Misc tab as we do for the All Scenery tab. As scenery row tabs are added, the Misc tab won't always be all the way on the right, so it can be hard to find at first. This will ensure it's always in the same place.

This change also has rows after the first one fully fill up with tabs, so there isn't a blank gap on the right, and each row does actually have 20 items in it.

image

(The PR had to be re-created because I messed up my local branch name. Sorry! [The original PR can be found here.](https://github.com/OpenRCT2/OpenRCT2/pull/19536))

### CODE REVIEW

**Comment by karst** [comment]
As I have mentioned in the previous PR as well. I feel stickying the misc tab to all scenery makes the misc tab pointless. All random scenery will be in both, making misc basically useless to click on.

**Comment by fidwell** [comment]
> As I have mentioned in the previous PR as well. I feel stickying the misc tab to all scenery makes the misc tab pointless. All random scenery will be in both, making misc basically useless to click on.

yeah I can understand that viewpoint. if the devs decide this isn't an improvement, I won't be offended

**Comment by ocalhoun6** [comment]
> All random scenery will be in both, making misc basically useless to click on.

_But_ the misc tab includes _only_ the scenery that's not in scenery groups, right? If the groupless scenery is specifically what you're looking for, then the misc tab makes it _much_ easier to find that.

**Comment by ZehMatt** [comment]
> As I have mentioned in the previous PR as well. I feel stickying the misc tab to all scenery makes the misc tab pointless. All random scenery will be in both, making misc basically useless to click on.

It seems to me there is still a fundamental difference between showing all scenery and only scenery that is uncategorized, the difference of the list can be quite huge and for consistency reasons I think this PR is sensible.

File: src/openrct2-ui/windows/Scenery.cpp [hunk 1]
```diff
@@ -1223,7 +1232,7 @@ class SceneryWindow final : public Window
 
     int32_t GetTabRowCount()
     {
-        int32_t tabEntries = static_cast<int32_t>(_tabEntries.size() - 1);
+        int32_t tabEntries = static_cast<int32_t>(_tabEntries.size());
         return std::max<int32_t>((tabEntries + MaxTabsPerRow - 1) / MaxTabsPerRow, 0);
```
Comment (Broxzier) [inline src/openrct2-ui/windows/Scenery.cpp:1236]:
```suggestion
        return std::max<int32_t>((_tabEntries.size() + MaxTabsPerRow - 1) / MaxTabsPerRow, 0);
```

**Comment by Broxzier** [comment]
This change seems fine with me, the misc tab is a somewhat special case anyway. I've tested this with parks that have and have no misc tab, and have different amounts of rows. All seem to work fine. Please rebase this for the changelog conflict.

**Comment by fidwell** [comment]
@Broxzier I've rebased, in case you missed it

**Pull request merged**